integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java

```diff
@@ -0,0 +1,280 @@
+      UUID testUuid = UUID.randomUUID();
+      log.info("TestUuid is {}", testUuid);
+
+      Path derbySystemHome =
+          userDirPath.resolve("tmp").resolve("derby").resolve(testUuid.toString());
+      Path sparkSqlWarehouse =
+          userDirPath.resolve("tmp").resolve("spark-sql-warehouse").resolve(testUuid.toString());
+      Path checkpointsDir =
+          userDirPath.resolve("tmp").resolve("checkpoints").resolve(testUuid.toString());
+
+      int randomPort = new Random().nextInt(1000) + 10000;
+
+      OpenLineageEndpointHandler handler = new OpenLineageEndpointHandler();
+      HttpServer server = HttpServer.create(new InetSocketAddress(randomPort), 0);
+      server.createContext("/api/v1/lineage", handler);
+      server.setExecutor(null);
+      server.start();
+
+      SparkSession spark =
+          SparkSession.builder()
+              .appName("kafka-source-kafka-sink")
+              .master("local[*]")
+              .config("spark.extraListeners", OpenLineageSparkListener.class.getCanonicalName())
+              .config("spark.driver.host", "localhost")
+              .config("spark.driver.extraJavaOptions", "-Dderby.system.home=" + derbySystemHome)
+              .config("spark.sql.warehouse.dir", sparkSqlWarehouse.toString())
+              .config("spark.ui.enabled", false)
+              .config("spark.openlineage.transport.type", "http")
+              .config("spark.openlineage.transport.url", "http://localhost:" + randomPort)
+              .config("spark.openlineage.facets.disabled", "[spark_unknown;]")
+              .getOrCreate();
+
+      Dataset<Row> sourceStream =
+          spark
+              .readStream()
+              .format("kafka")
+              .option("subscribe", "source-topic")
+              .option("kafka.bootstrap.servers", bootstrapServers)
+              .option("startingOffsets", "earliest")
+              .load();
+
+      StructType schema = StructType.fromDDL("id STRING, epoch LONG");
+      StreamingQuery streamingQuery =
+          sourceStream
+              .selectExpr("CAST(value AS STRING) AS value")
+              .select(from_json(col("value"), schema).as("event"))
+              .select(col("event.id").as("id"), col("event.epoch").as("epoch"))
+              .select(col("id"), from_unixtime(col("epoch")).as("timestamp"))
+              .select(functions.struct(col("id"), col("timestamp")).as("converted_event"))
+              .select(functions.to_json(col("converted_event")).as("value"))
+              .writeStream()
+              .format("kafka")
+              .option("topic", "target-topic")
+              .option("kafka.bootstrap.servers", bootstrapServers)
+              .option("checkpointLocation", checkpointsDir.toString())
+              .trigger(Trigger.ProcessingTime(Duration.ofSeconds(4).toMillis()))
+              .start();
+
+      streamingQuery.awaitTermination(Duration.ofSeconds(20).toMillis());
+
+      spark.stop();
```

Comment:
Oftentimes streaming jobs run for a long time and it makes sense to emit more events than only START and SUCCESS/FAIL. For flink, we're sending RUNNING events every configured time interval. This can make sense for Spark as well. If so, pls create a follow-up issue.

Reply:
RUNNING events are also there, I can create follow up issue with events sent with specific time interval.